### ISSUE TITLE
[feat] Add more licensing options for minting

### ISSUE THREAD

**Issue by LeonNicholls** [teia-community/teia-ui#161]
### Describe the problem

There are currently six licensing options (excluding custom) on the minting form. Consider adding more licensing options to cover public domain and open source licenses.

### Describe the solution you'd like

We should add a few popular ones and maybe add others on-demand:
- GNU GPL
- MIT
- CC0others

**Comment by melMass** [comment]
Thanks for filling the issue, I'm wondering if we should maybe upload and pin all the licenses (their content) on IPFS and use the URI of the ones selected from the UI, briefly reading the spec I guess it makes sense. Wdyt?

**Comment by LeonNicholls** [comment]
Having a copy of the common licenses isn't necessary since they are easily publicly available, and maybe an HTTP URL will be fine. But this is an interesting use of IPFS as an alternative.

**Comment by melMass** [comment]
> 

Yes, IMO they need to be added to the NFT itself and not only by name, and HTTP URL instead of IPFS could be fine but it avoids link breaks etc and we can just preupload the ones from the dropdown and automatically add the IPFS link. Can you provide a http link of those licenses?

My idea was to take the ones from here: https://github.com/licenses/license-templates.git
Strip name and year and upload each to IPFS and just add a map to each files and the license values

**Comment by melMass** [comment]
Or we can actually feed the templates with data from the user and upload it for them each time too?
Might be more compliant as some license require those infos

**Comment by LeonNicholls** [comment]
Are you saying if the user specifies an HTTP URL to a license, we will automatically upload a copy of that content to IPFS and then use that IPFS URI in the metadata instead of the user's HTTP URL?

**Comment by melMass** [comment]
> Are you saying if the user specifies an HTTP URL to a license, we will automatically upload a copy of that content to IPFS and then use that IPFS URI in the metadata instead of the user's HTTP URL?

We could also do that, even if it's more prone to error.
I meant mapping the ones that are currently in the dropdown + the ones you listed in that issue to the licenses from the repo I linked (preuploaded for the ones that don't require input) and adding the required inputs depending on the selected license (organization name, author, years etc...)

I think that's "more web3"

**Comment by LeonNicholls** [comment]
Ok; understood. That repo has very old copies of the licenses. I know some of the open source licenses have run into legal issues over the years requiring updates. Probably best to go to the source of each of the licenses (which might not have the nice template placeholders).

**Comment by melMass** [comment]
> Ok; understood. That repo has very old copies of the licenses. I know some of the open source licenses have run into legal issues over the years requiring updates. Probably best to go to the source of each of the licenses (which might not have the nice template placeholders).

That's not a big deal, I asked over discord if someone could gather them, but I will just do it, I have something basic in mind that will be a good starting point to iterate upon.

**Comment by Zir0h** [comment]
I think this one can be closed after all of @ryangtanaka-org 's work?

**Comment by LeonNicholls** [comment]
Yes, although its not clear on the minting form that there is even a feature like: https://teia.art/copyright
We might have to look at the minting UX to tell artists about the copyright feature.
